cmd/payments/payments.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"context"
+	"encoding/csv"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/accounting"
+	"storj.io/storj/pkg/utils"
+	"storj.io/storj/satellite/satellitedb"
+)
+
+// generateCSV generates a payment report for all nodes for a given period
+func generateCSV(ctx context.Context, cfgDir string, dbPath string, id string, start time.Time, end time.Time) (string, error) {
+
+	db, err := satellitedb.New(dbPath)
+	if err != nil {
+		return "", errs.New("error connecting to master database on satellite: %+v", err)
+	}
+	defer func() {
+		err := db.Close()
+		if err != nil {
+			fmt.Printf("error closing connection to master database on satellite: %+v\n", err)
+		}
+	}()
+
+	pDir := filepath.Join(cfgDir, "payments")
+
+	if err := os.MkdirAll(pDir, 0700); err != nil {
+		return "", err
+	}
+
+	layout := "2006-01-02"
+	filename := id + "--" + start.Format(layout) + "--" + end.Format(layout) + ".csv"
+	path := filepath.Join(pDir, filename)
+	file, err := os.Create(path)
+	if err != nil {
+		return "", err
+	}
+	defer utils.LogClose(file)
```

Comment:
Do not use LogClose for new code.

Reply:
Is there a new we are closing and checking errors?

Reply:
file.Close and errs.Combine or wrap depending on the situation